File: cpp/bc3301.cpp
#include <iostream>

using namespace std;

int main()
{
	int h, z, l;
	string res = "";
	
	cin >> h >> z >> l;
	
	bool fVe, sVe;
	
	fVe = (h > z && h < l) || (h < z && h > l);
	sVe = (z > h && z < l) || (z < h && z > l);
	
	
	if(fVe == true) res += "huguinho";
	else if(sVe == true) res += "zezinho";
	else res += "luisinho";
	
	cout << res << endl;
}

File: python/bc3301.py
res = ""
cVal = input().split(" ")

h = int(cVal[0])
z = int(cVal[1])
l = int(cVal[2])

if (h > z and h < l) or (h < z and h > l): res = "huguinho"
elif (z > h and z < l) or (z < h and z > l): res = "zezinho"
else: res = "luisinho"

print(res)
